### DISCUSSION TITLE
Table with virtual list

### DISCUSSION THREAD

**Discussion by oemueller** [SchwarzIT/onyx#451] [Ideas]
To support really high numbers of rows in the table component one could implement a
virtual list / scrolling. 
Without that approach the performance suffers sooner or later.

Example approaches:
https://github.com/Akryum/vue-virtual-scroller
https://vueuse.org/core/useVirtualList/

**Comment by mj-hof** [comment]
Hi @oemueller. We consider your idea further and will wait for more feedback. We assume that this issue is covered by our pagination and lazy loading approach 😄

**Comment by EnricoMessall** [comment]
Hey. We introduced a OnyxScroller component in our project to fit our requirements of a virtual scroller for lazy loading a list. Maybe that can be useful. It makes use of the scroll target as described in https://github.com/SchwarzIT/onyx/discussions/4516 also:

```ts
<script lang="ts" setup generic="TModelValue extends any[], TScrollValue extends TModelValue[number]">
import { OnyxLoadingIndicator } from "sit-onyx";
import { computed, ref } from "vue";

interface ScrollerProps {
  modelValue: TModelValue;
  entryKey?: "value" | string;
  gap?: string | undefined;
  lazy?: boolean;
  loading: boolean;
  scrollOffset?: number;
  maxHeight?: string;
  scrollTarget?: HTMLElement | null;
}

const scroller = ref<HTMLElement | null>(null);

const props = withDefaults(defineProps<ScrollerProps>(), {
  lazy: false,
  entryKey: "value",
  scrollOffset: 300,
  scrollTarget: null,
});

const activeScrollTarget = computed(() => {
  return props.scrollTarget || scroller.value;
});

const emit = defineEmits<LazyLoadingEmits>();

const _slots = defineSlots<{
  /**
   * Optional slot to override the entry content.
   */
  entry?(props: TScrollValue): unknown;
}>();

const { vScrollEnd, isScrollEnd } = useScrollEnd({
    enabled: computed(() => props.lazy || false),
    loading: computed(() => props.loading),
    offset: computed(() => props.scrollOffset),
    target: activeScrollTarget,
  });

  watchEffect(() => {
    if (isScrollEnd.value) {
      emit("lazyLoad");
    }
  });
</script>

<template>
  <div>
    <ul ref="scroller" v-scroll-end class="scroller" :style="{ 'max-height': maxHeight, gap: gap }">
      <li v-for="entry in props.modelValue" :key="entry[entryKey]" class="scroller-item">
        <slot name="entry" v-bind="entry">
          {{ entry.label }}
        </slot>
      </li>
    </ul>
    <OnyxLoadingIndicator v-if="loading" type="dots" />
  </div>
</template>

<style scoped>
.scroller {
  display: flex;
  flex-direction: column;
  padding: 0;
  list-style-type: none;
  overflow-y: auto;
}
</style>
```

**Comment by JoCa96** [comment]
Hi @oemueller @EnricoMessall, thanks a lot for your input and suggestions. We would like to keep rows with their data in the DOM, so the stay discoverable and interactive.

We are aware of virtual scrolling and will keep it in mind. For now we will put this optimization off and prioritize less disruptive forms of optimization.

Don't let this discourage you from creating further discussions, suggestions, and issues! We are happy about any input you might have! 😊
